pkg/operator/ceph/webhook.go

```diff
@@ -18,47 +18,102 @@ package operator
 	"github.com/pkg/errors"
 	"github.com/rook/rook/pkg/clusterd"
 	"github.com/rook/rook/pkg/operator/k8sutil"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/version"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 const (
 	admissionControllerAppName       = "rook-ceph-admission-controller"
 	tlsPort                    int32 = 443
+	minK8sVersionWebhook             = "1.16.0"
+	webhookEnv                       = "ROOK_DISABLE_ADMISSION_CONTROLLER"
 )
 
 var (
-	namespace = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	namespace              = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	certManagerWebhookName = "cert-manager-webhook"
 )
 
-func isSecretPresent(ctx context.Context, context *clusterd.Context) (bool, error) {
+func createWebhook(ctx context.Context, context *clusterd.Context) (bool, error) {
+	k8sVersion, err := k8sutil.GetK8SVersion(context.Clientset)
+	if err != nil {
+		logger.Errorf("failed to get k8s version. %v", err)
+		return false, nil
+	}
+	if !k8sVersion.AtLeast(version.MustParseSemantic(minK8sVersionWebhook)) {
+		logger.Errorf("minimum versions %s required for webhook to run", err)
+		return false, nil
+	}
+
+	admission, err := cs.NewForConfig(context.KubeConfig)
+	if err != nil {
+		logger.Errorf("failed to set config for cert-manager. %v", err)
+		return false, nil
+	}
+
+	if os.Getenv(webhookEnv) == "true" {
+		logger.Info("delete Issuer and Certificate since secret is not found")
+		if err = deleteIssuerAndCetificate(ctx, admission, context); err != nil {
+			logger.Errorf("failed to delete issuer or certificate. %v", err)
+		}
+		return false, nil
+	}
+
+	logger.Infof("Fetching webhook %s to see if cert-manager is installed.", certManagerWebhookName)
+	_, err = context.Clientset.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, certManagerWebhookName, metav1.GetOptions{})
+	if err != nil {
+		logger.Info("failed to get cert manager")
+		return false, nil
+	}
+
+	issuer, err := fetchorCreateIssuer(ctx, admission)
+	if err != nil {
+		logger.Errorf("issuer creation failed %v", err)
+		return false, nil
+	}
+
+	err = fetchorCreateCertificate(ctx, admission, issuer)
+	if err != nil {
+		logger.Errorf("certificate creation failed %v", err)
+		return false, nil
+	}
+
 	logger.Infof("looking for admission webhook secret %q", admissionControllerAppName)
 	s, err := context.Clientset.CoreV1().Secrets(namespace).Get(ctx, admissionControllerAppName, metav1.GetOptions{})
 	if err != nil {
 		// If secret is not found. All good ! Proceed with rook without admission controllers
+		logger.Info("delete Issuer and Certificate since secret is not found")
+		if err = deleteIssuerAndCetificate(ctx, admission, context); err != nil {
+			logger.Errorf("failed to delete issuer or certificate. %v", err)
```

Comment:
It's going to be normal to fail to delete because you will probably get a not found error, so maybe an info level is fine
```suggestion
			logger.Infof("could not delete issuer or certificate. %v", err)
```